src/pages/HostingPreferences.vue

```diff
@@ -21,35 +21,81 @@ import PrimaryLayout from 'components/PrimaryLayout.vue'
 import HAppSelectionSection from 'components/settings/hostingPreferences/HAppSelectionSection'
 import InvoicesSection from 'components/settings/hostingPreferences/InvoicesSection.vue'
 import PricesSection from 'components/settings/hostingPreferences/PricesSection.vue'
-import { reactive } from 'vue'
+import { usePreferencesStore } from '@/store/preferences'
+import { reactive, computed, onMounted} from 'vue'
 
-const invoicesSettings = reactive({
+const preferencesStore = usePreferencesStore()
+
+const localInvoicesSettings = reactive({
   frequency: {
     amount: 100,
     period: 7
   },
   due: {
     period: 7
-  }
+  },
 })
 
-const pricesSettings = reactive({
+const localPricesSettings = reactive({
   cpu: 100,
   storage: 100,
-  bandwidth: 100
+  bandwidth: 100,
 })
 
 function updateFrequency(value) {
-  invoicesSettings.frequency = value
+  localInvoicesSettings.frequency = value
 }
 
 function updateDue(value) {
-  invoicesSettings.due = value
+  localInvoicesSettings.due = value
 }
 
 function updatePrice({ prop, value }) {
-  pricesSettings[prop] = value
+  localPricesSettings[prop] = value
 }
+
+const pricesSettings = computed(() => ({
+  ...localPricesSettings,
+  ...preferencesStore.pricesSettings
+}))
+
+const invoicesSettings = computed(() => ({
+  ...localInvoicesSettings,
+  ...preferencesStore.invoicesSettings
+}))
+
```

Comment:
@mateuszRybczonek  - This part is more a demonstration of an idea than a hardline implementation.  It's here to underline the point that we prob need to keep some local state reference making the call to allow for the user to play with/locally update the prices before submitting the call to set the host invoice/price preferences - as this can only be done once.